free-programming-books.md

```diff
@@ -379,6 +380,7 @@
 
 * [Competitive Programmer's Handbook](https://cses.fi/book.html) - Antti Laaksonen (PDF)
 * [Competitive Programming, 1st Edition](https://cpbook.net/#CP1details) (PDF)
+* [Confident Programmer Problem Solver: Six Steps Programming Students Can Take To Solve Coding Problems](https://www.smashwords.com/books/view/1001889) [(PDF, EPUB, MOBI)]
```

Comment:
not free

Reply:
I have replaced it with another one. Sorry for the inconvenience.

Reply:
add author name

Reply:
Sir,

I have resubmitted the PR with author name added to the book "Principles of Algorithmic Problem Solving". I removed "Confident Programmer Problem Solver" book since it was not free as you rightly pointed it out.

The goalkicker book on Algorithms Notes for Professionals was compiled from StackOverflow Documentation and there is no individual author.

Thank you.